### PR TITLE
Optimized for size Mathf.pow

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

File: std/assembly/math.ts [hunk 1]
```diff
@@ -2592,7 +2632,40 @@ export namespace NativeMathf {
       if (y == 1.0) return x;
       if (y == 0.0) return 1.0;
     }
-    return powf_lut(x, y);
+    if (ASC_SHRINK_LEVEL < 1) {
+      // see: musl/src/math/powf.c
+      return powf_lut(x, y);
+    } else {
+      // based on: metallic/src/math/float/powf.c
```
Comment (dcodeIO) [inline std/assembly/math.ts:2639]: The file linked here, [metallic/src/math/float/powf.c](https://github.com/jdh8/metallic/blob/master/src/math/float/powf.c), doesn't appear to resemble the following code. Am I looking at the wrong file or is this expected?

File: std/assembly/math.ts [hunk 2]
```diff
@@ -1880,6 +1880,46 @@ function tan_kernf(x: f64, odd: i32): f32 { // see: musl/tree/src/math/__tandf.c
   return <f32>(odd ? -1 / r : r);
 }
 
+// @ts-ignore: decorator
+@inline
+function log2f(x: f64): f64 {
+  const
```
Comment (dcodeIO) [inline std/assembly/math.ts:1886]: Perhaps reference `metallic/src/math/float/log2f.c`.

File: std/assembly/math.ts (hunk 1 above)
Comment (MaxGraey) [reply std/assembly/math.ts:2639]: It not exactly the same but it based on metallic version with manually inlined and slightly reworked part for handling special cases

File: std/assembly/math.ts [hunk 3]
```diff
@@ -1880,6 +1880,46 @@ function tan_kernf(x: f64, odd: i32): f32 { // see: musl/tree/src/math/__tandf.c
...
+  x = (x - 1) / (x + 1);
+  var xx = x * x;
+  var y = x * (c0 + c1 * xx + (c2 + c3 * xx) * (xx * xx));
+  return (2 * log2e) * y + <f64>exponent;
+}
+
+// @ts-ignore: decorator
+@inline
+function exp2f(x: f64): f64 {
+  const
```
Comment (dcodeIO) [inline std/assembly/math.ts:1905]: Is this based on metallic?

Comment (MaxGraey) [reply std/assembly/math.ts:1907]: yeah

Comment (dcodeIO) [reply std/assembly/math.ts:1907]: Code below seems to be based on metallic/math/float/exp2f.h, so with exp2f.c I indeed looked at a wrong file, which is now the comment here. I'm confused :)

File: std/assembly/math.ts (hunk 2 above)
Comment (dcodeIO) [reply std/assembly/math.ts:1887]: And this seems to be rather `metallic/math/float/finite/log2f.h`. Quite confusing structure over there.

File: std/assembly/math.ts (hunk 3 above)
Comment (MaxGraey) [reply std/assembly/math.ts:1907]: Added more paths

**Comment by dcodeIO** [comment]
Speaking of paths, do you know if we are testing all possible combinations? Like currently we instantiate and run untouched and fully optimized, but nothing in between, which might make us miss something for certain ASC_XY levels.

**Comment by MaxGraey** [comment]
`untouched` usually cover `ASC_SHRINK_LEVEL < X` and `optimized` `ASC_SHRINK_LEVEL >= X` so I expect all condition compile paths covered

**Pull request merged**